File: Backend/src/main/java/org/example/backend/dto/TeacherPreferredTimeDto.java
package org.example.backend.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Value;

import java.io.Serializable;

/**
 * DTO for {@link org.example.backend.domain.TeacherPreferredTime}
 */
@Data
@NoArgsConstructor
public class TeacherPreferredTimeDto implements Serializable {
    TeacherPreferredTimeIdDto id;
    TeacherDto teacher;
    PreferredTimeDto preferredTime;
}

File: Frontend/src/app/dto/teacher-preferred-time-dto.ts
import {TeacherPreferredTimeIdDto} from "./teacher-preferred-time-id-dto";
import {TeacherDto} from "./teacher-dto";
import {PreferredTimeDto} from "./preferred-time-dto";

export interface TeacherPreferredTimeDto {
  id : TeacherPreferredTimeIdDto
  teacher : TeacherDto
  preferredTime : PreferredTimeDto
}
